### PR TITLE
Gh page markdown link check [skip ci]

### PR DESCRIPTION
**Pull request by nvliyuan**
add github action to check broken markdown links.
related to #618

### CODE REVIEW

**Comment by nvliyuan** [comment]
we use [the tool](https://github.com/marketplace/actions/markdown-link-check ) to enable markdown link check github action, if all links are good, we could create a pull request, if not, the pull request will be blocked.
image
image

**Comment by nvliyuan** [comment]
I am not sure why https://mvnrepository.com/artifact/com.nvidia/rapids-4-spark_2.12 is regarded as a dead link by this tool, so I add `"pattern": "https:\/\/mvnrepository"` as ignore pattern to workaround.

**Comment by nvliyuan** [comment]
we can also add the same github action to dev-branch 22.04 see #4907

File: .github/workflows/markdown-links-check.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        max-depth: -1
+        use-verbose-mode: 'yes'
+        config-file: '.github/workflows/markdown-links-check/markdown-links-check-config.json'
```
Comment (zhanga5) [inline .github/workflows/markdown-links-check.yml:14]: `nitpick`: no newline at end of file

**Comment by jlowe** [comment]
> we can also add the similar github action to dev-branch 22.04 see https://github.com/NVIDIA/spark-rapids/pull/4907

We do not develop on gh-pages directly.  All changes on gh-pages should be developed first on a dev branch (e.g.: branch-22.04) and then merged/cherry-picked from there.

**Comment by nvliyuan** [comment]
> > we can also add the similar github action to dev-branch 22.04 see #4907
> 
> We do not develop on gh-pages directly. All changes on gh-pages should be developed first on a dev branch (e.g.: branch-22.04) and then merged/cherry-picked from there.

yes, but gh-pages branch has slightly different from the dev-branch because it doesn't have code, so if we merge/cherry-picked some specific file like` rapids-udfs.md`(like https://github.com/NVIDIA/spark-rapids/blob/branch-22.04/docs/additional-functionality/rapids-udfs.md?plain=1#L55),  it will break some hyperlink in the [doc](https://nvidia.github.io/spark-rapids/docs/additional-functionality/rapids-udfs.html). In order to prevent similar issues, should we also add this markdown-check GitHub action to gh-pages branch?

File: .github/workflows/markdown-links-check.yml (hunk 1 above)
Comment (nvliyuan) [reply .github/workflows/markdown-links-check.yml]: add a newline

File: .github/workflows/markdown-links-check/markdown-links-check-config.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "timeout": "5s",
+  "retryCount":10,
+  "aliveStatusCodes": [403, 400, 302, 200, 0]
```
Comment (jlowe) [inline .github/workflows/markdown-links-check/markdown-links-check-config.json:4]: This is getting out of sync with #4907.  I'd prefer we focus on #4907, get that merged, then merge that commit to gh-pages with a change for the `base-branch` during that PR.  Doing this as parallel PRs is leading to diverging code and likely one will have bugs/features the other will not.

Comment (nvliyuan) [reply .github/workflows/markdown-links-check/markdown-links-check-config.json:4]: sure, closed this pr,  thanks

**Comment by nvliyuan** [comment]
focus on #4907 and close this pr

**Pull request closed without merging**